internal/api/dispatch/impl.go

```diff
@@ -106,3 +107,40 @@ func (this *dispatchManager) ProcessRun(ctx context.Context, account string, ser
 	instrumentation.RunCreated(ctx, run.Recipient, entity.ID, run.Url, entity.Service, protocol.GetLabel())
 	return entity.ID, correlationID, nil
 }
+
+func (this *dispatchManager) ProcessCancel(ctx context.Context, account string, cancel generic.CancelInput) (runID, correlationID uuid.UUID, err error) {
+	var run db.Run
+
+	protocol := *protocols.SatelliteProtocol
```

Comment:
We cannot assume this. It's possible for the caller to pass any run id including a run id for non-Satellite run. We'll need to first validate that the given run id belongs to a sat run and reject the request if it does not.